test/e2e/network/conntrack.go

```diff
@@ -103,16 +103,18 @@ var _ = SIGDescribe("Conntrack", func() {
 				len(nodes.Items))
 		}
 
-		ips := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
+		ips := e2enode.GetAddresses(&nodes.Items[0], v1.NodeInternalIP)
 
 		clientNodeInfo = nodeInfo{
 			name:   nodes.Items[0].Name,
 			nodeIP: ips[0],
 		}
 
+		ips = e2enode.GetAddresses(&nodes.Items[1], v1.NodeInternalIP)
+
 		serverNodeInfo = nodeInfo{
 			name:   nodes.Items[1].Name,
-			nodeIP: ips[1],
+			nodeIP: ips[0],
```

Comment:
we still need to be sure that the IP we choose correspond to the main IP family the cluster is using, `framework.TestContext.ClusterIsIPv6()`

Reply:
Is that necessary since we are doing a separate hardcoded check of GetAddresses? It works locally on a dualstack cluster.

Reply:
you don't control the order of the internalIP returned, and if the main family is IPv4 and the ip returned is IPv6 it will fail , actually is the reason why the tests fail today, it connects to an IPv4 service (by default services are SingleStack on the main family) but uses the IPv6 node address to connect to it ...

```

ï¿½[1mSTEPï¿½[0m: waiting up to 3m0s for service svc-udp in namespace conntrack-2577 to expose endpoints map[pod-server-1:[80]]
Nov  3 15:13:54.111: INFO: successfully validated that service svc-udp in namespace conntrack-2577 exposes endpoints map[pod-server-1:[80]]
ï¿½[1mSTEPï¿½[0m: checking client pod connected to the backend 1 on Node IP fc00:f853:ccd:e793::4
Nov  3 15:14:54.143: INFO: Pod client logs: Tue Nov  3 15:13:49 UTC 2020
Tue Nov  3 15:13:49 UTC 2020 Try: 1
```

Reply:
Ah ok, I thought it was just failing bc we also picked the 2nd entry which consistently ended up being the IPv6 address, but that makes sense now.